app/models/ebook.rb

```diff
@@ -42,6 +46,8 @@ class Ebook < ApplicationRecord
     presence: true
 
   scope :live, -> { where(status: :live) }
+  scope :filter_by_tags, ->(tag_ids) { joins(:ebook_tags).where(ebook_tags: { tag_id: tag_ids }) }
+  scope :filter_by_users, ->(user_ids) { where(user_id: user_ids) }
```

Comment:
Estas filtragens estão a ir buscar apenas ebooks que tenham **todas** as tags passadas como argumentos. Além disso, posso estar enganado, mas como estas a fazer um join, esta query vai eliminar ebooks que não tenham tags nenhumas. É suposto?

Reply:
Na verdade a query está a retornar as tags que tenham um id pertencente ao conjunto de ids passados. E.g. `SELECT "tags".* FROM "tags" WHERE "tags"."id" IN (?, ?)  [["id", 9], ["id", 3]]`. Ou seja, para as tags "Western" e "Fiction" retorna os ebooks que tenham a tag "Western" **e/ou** "Fiction". Quando falámos não tinha a certeza mas está a fazer o que se pretende :)